odoo/http.py

```diff
@@ -1042,28 +1042,24 @@ def authenticate(self, dbname, credential):
             'HTTP_HOST': request.httprequest.environ['HTTP_HOST'],
             'REMOTE_ADDR': request.httprequest.environ['REMOTE_ADDR'],
         }
-
-        registry = Registry(dbname)
-        auth_info = registry['res.users'].authenticate(dbname, credential, wsgienv)
+        env = env(user=None, su=False)
+        auth_info = env['res.users'].authenticate(credential, wsgienv)
         pre_uid = auth_info['uid']
 
         self.uid = None
         self.pre_login = credential['login']
         self.pre_uid = pre_uid
 
-        with registry.cursor() as cr:
-            env = odoo.api.Environment(cr, pre_uid, {})
+        env = env(user=pre_uid)
 
-            # if 2FA is disabled we finalize immediately
-            user = env['res.users'].browse(pre_uid)
-            if auth_info.get('mfa') == 'skip' or not user._mfa_url():
-                self.finalize(env)
+        # if 2FA is disabled we finalize immediately
+        user = env['res.users'].browse(pre_uid)
+        if auth_info.get('mfa') == 'skip' or not user._mfa_url():
+            self.finalize(env)
 
-        if request and request.session is self and request.db == dbname:
-            request.env = odoo.api.Environment(request.env.cr, self.uid, self.context)
+        if request and request.session is self:
+            request.env = env(user=self.uid, context=self.context)
```

Comment:
so we also update `threading.current_thread().uid`
```suggestion
            request.update_env(user=self.uid, context=self.context)
```

Reply:
In the end, this change makes 2 unit test failing
- https://runbot.odoo.com/runbot/build/69946316
- https://runbot.odoo.com/runbot/build/69946717

Should I revert to as before, to keep a minimum diff and keep the behavior as much as it was before, or do you want to handle the case ? The problem lies on the fact  `update_env` doesn't work if `request` has no `env` in the first place.

Reply:
Ahhhh yes... that method is used to grant an environment on request... sometime. You can change it back to `request.env = ...`. Sorry for the inconvenience!